source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -37,6 +39,11 @@ bool RoleBasedAccessControlEngineImpl::allowed(const Network::Connection& connec
   return matched == allowed_if_matched_;
 }
 
+bool RoleBasedAccessControlEngineImpl::allowed(const Network::Connection& connection) const {
+  return allowed(connection, Envoy::Http::HeaderMapImpl(), envoy::api::v2::core::Metadata(),
```

Comment:
make the HeaderMapImpl and Metadata static variable? HeaderMapImpl does memory allocations for inline headers, constructing everytime is not good. Ideally we should make the parameter a pointer instead of a reference.

Reply:
Done.